scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_presenter.py

```diff
@@ -277,6 +290,58 @@ def plot_default_groups_or_pairs(self):
             else:  # else plot groups
                 self.grouping_table_widget.plot_default_case()
 
+    def handle_period_information_button_clicked(self):
+        if self._model.is_data_loaded() and self.period_info_widget.is_empty():
+            self._add_period_info_to_widget()
+        self.period_info_widget.show()
+        self.period_info_widget.raise_()
+
+    def _add_period_info_to_widget(self):
+        runs = self._model._data.current_runs
+        runs_string = ""
+        for run_list in runs:
+            for run in run_list:
+                if runs_string:
+                    runs_string += ", "
+                runs_string += str(run)
+        self.period_info_widget.set_title_runs(self._model.instrument + runs_string)
+        period_sequences_log = self._model._data.get_sample_log("period_sequences")
+        self.period_info_widget.number_of_sequences = str(period_sequences_log.value) if period_sequences_log else None
+        names_log = self._model._data.get_sample_log("period_labels")
+        names = names_log.value.split(";") if names_log else []
+        types_log = self._model._data.get_sample_log("period_type")
+        types = types_log.value.split(";") if types_log else []
+        frames_log = self._model._data.get_sample_log("frames_period_requested")
+        frames = frames_log.value.split(";") if frames_log else []
+        total_frames_log = self._model._data.get_sample_log("frames_period_raw")
+        total_frames = total_frames_log.value.split(";") if total_frames_log else []
+        counts_log = self._model._data.get_sample_log("total_counts_period")
+        counts = counts_log.value.split(";") if counts_log else []
+        tags_log = self._model._data.get_sample_log("period_output")
+        tags = tags_log.value.split(";") if tags_log else []
```

Comment:
Ditto as above, remove some code duplication here:
```
def _add_period_info_to_widget(self:

    def _parse_logs(log_name):
          log_value = self._model._data.get_sample_log("period_labels")
          return log_value if log_value else []

    names_log = _parse_logs("period_labels")
    types_log = _parse_logs("period_type")
    # etc.
```

We could write a fancy loop that does `for i in ["period_labels", "period_type"...]` etc. but it starts getting into "too smart for it's own good"